feat: add counts from vespa to geo pages

jamesgorrie opened the pull request:
# What's changed
- adds a `api/geography-counts` endpoint to be access by the geography page that gets it's data from multiple vespa seaches
- enable vespa to return page_size, in this case `0`
- small refactor of the geos page
- gives themes an index so we can access them easier

## Why?

[Described here](https://linear.app/climate-policy-radar/issue/APP-1066/use-document-type-from-search-as-the-grouped-lists-on-geography-pages).

I was trying to run multiple fetches from the client, and I was seeing some pretty gnarly performance problems. This shifts it to the Node server, and I am seeing general results of about ~200ms. 

## diagrams
```mermaid
flowchart LR
    GeographyPage --> ThemeCategories --> NextJsApi["/api/geography-counts"] --> MultiVespaQueriesPageSize0
    MultiVespaQueriesPageSize0 --> GeographyPage
```

jamesgorrie commented on src/components/pages/geographyOriginalPage.tsx:
```diff
@@ -209,43 +220,56 @@ export const GeographyOriginalPage = ({ geographyV2, summary, targets, theme, th
-  const handleVespaSearchTabClick = async (e: React.MouseEvent<HTMLButtonElement, MouseEvent>, index: number, value: string) => {
-    const selectedThemeCategory = vespaSearchTabbedNavItems.find((category) => category.title === value);
-    const categoryFilter = selectedThemeCategory.slug;
+  const countCategories = vespaSearchTabbedNavItems.map((item) => item.slug.toLocaleLowerCase()).filter((slug) => slug !== "litigation");
```
Not massively sure. Currently it breaks as the [config is actually invalid](https://github.com/climatepolicyradar/navigator-frontend/blob/main/themes/cpr/config.ts#L90). I am hacking around that hack

PatFawbertMills replied:
We are not enabling litigation on CPR app until we resolve supporting all types of families with the new designs.

PatFawbertMills replied:
Which I would be surprised if we have done for CCC launch

jamesgorrie commented on src/components/pages/geographyOriginalPage.tsx:
```diff
@@ -209,43 +220,56 @@ export const GeographyOriginalPage = ({ geographyV2, summary, targets, theme, th
   };
 
   /** Vespa search results */
+  const [counts, setCounts] = useState({});
```
100%!

jamesgorrie commented on src/components/pages/geographyOriginalPage.tsx:
```diff
@@ -209,43 +220,56 @@ export const GeographyOriginalPage = ({ geographyV2, summary, targets, theme, th
+      .then((data) => setCounts(data.counts));
+    // We onnly ever want this to run once
+    /* trunk-ignore(eslint/react-hooks/exhaustive-deps)*/
+  }, []);
```
> Surely you'd want to refetch the stats if it does?

The whole component would reload at that point, which is fine. I'll have a go at trying to work out why the suggested deps cause a cycle, but always feel like I'm chasing a linter, not good code or value when I do.

jamesgorrie commented:
@PatFawbertMills @annaCPR thanks for all the great comments and suggestions! Will definitely get round to them. 

What do you think of the
- get critical, initial data via server
- render
- get less critical, data, which adds ~250ms to the request on the client
- render that

Part of this?

The reason to do this is if we defer this all to the server, our initial response is going to be slower. 

We could get all the searches at once - but this feels like an even more non performant solution, and for content that I imagine isn't looked at too often (we need to get tracking on these pages!)

jamesgorrie commented:
@PatFawbertMills @annaCPR I've added tests and updated the typing.

I have left the deps on the useEffect as it works and the code is representative of what we want to happen.

I can retrospectively go back and work out why adding the deps causes a infinite cycle.

**But given that this is creating a load of toil and the fact we need to keep responding to issues make me think the value that would add is blocking more value in freeing up engineering time**

PatFawbertMills reviewed (approved):
Makes sense, thanks, some of this we will revisit when we merge the other apps into the new page as well
